### ISSUE TITLE
sundials@develop issue (with superlu-dist) on macos

### ISSUE THREAD

**Issue by balay** [xsdk-project/xsdk-issues#142]
[spack-build-out.txt](https://github.com/xsdk-project/xsdk-issues/files/6711040/spack-build-out.txt)

```
ref ref: balay/xsdk-develop  c945a684b2714c332f2c8db9eac6e554e86cc23c
xsdk@kpro spack-xsdk % python3 ./bin/spack install --fail-fast -j6 xsdk@develop+trilinos~butterflypack~heffte~datatransferkit%apple-clang@12.0.0 fflags=-fallow-argument-mismatch
```

**Comment by balos1** [comment]
This appears to be an issue in SUNDIALS. [We check](https://github.com/LLNL/sundials/blob/master/include/sunlinsol/sunlinsol_superludist.h#L43.) for the SuperLU DIST version. Since SuperLU DIST @ develop is now a new major version, the check fails.

**Comment by balos1** [comment]
This patch should solve the problem. Instead of adding it as a patch in spack, I will see if we can change the sundials@develop target to be something newer that includes this.

```
diff --git a/include/sunlinsol/sunlinsol_superludist.h b/include/sunlinsol/sunlinsol_superludist.h
index 34512c3f2..80c9b8c66 100644
--- a/include/sunlinsol/sunlinsol_superludist.h
+++ b/include/sunlinsol/sunlinsol_superludist.h
@@ -40,7 +40,7 @@
 #include <sundials/sundials_nvector.h>
 #include <sunmatrix/sunmatrix_slunrloc.h>
 
-#if (SUPERLU_DIST_MAJOR_VERSION >= 6) && (SUPERLU_DIST_MINOR_VERSION >= 3)
+#if (SUPERLU_DIST_MAJOR_VERSION >= 7) || ((SUPERLU_DIST_MAJOR_VERSION == 6) && (SUPERLU_DIST_MINOR_VERSION >= 3))
 #define xLUstructInit dLUstructInit
 #define xScalePermstructInit dScalePermstructInit
 #define xScalePermstructFree dScalePermstructFree
```

**Comment by balay** [comment]
@balos1 Thanks! I've added this to `balay/xsdk-develop` for now.

https://github.com/spack/spack/commit/d94d1e413751551fdbfdc975bcab2f584cdf10c8

Looks like `remove-links-to-OpenMP-vector.patch` is also missing from current sundials@develop

**Comment by balos1** [comment]
Fixed in sundials 5.8.0.
